### PR TITLE
[objcruntime] Fix some `IntPtr` -> `NativeHandle` inside `Class`

### PR DESCRIPTION
**Pull request by spouliot**
* Using `NativeHandle` avoids implicit casts [1]
* Do not expose `IntPtr` as the type for the `SuperClass` handle

[1] likely not an issue with JIT/AOT optimization, less sure for the
interpreter. No harm in not having those in the product assembly.

### CODE REVIEW

File: src/ObjCRuntime/Class.cs [hunk 1]
```diff
@@ -65,8 +65,8 @@ public Class (string name)
 		{
 			this.handle = objc_getClass (name);
 
-			if (this.handle == IntPtr.Zero)
-				throw new ArgumentException (String.Format ("'{0}' is an unknown class", name));
+			if (handle == NativeHandle.Zero)
+				ObjCRuntime.ThrowHelper.ThrowArgumentException (nameof (name), "Unknown class");
```
Comment (rolfbjarne) [inline src/ObjCRuntime/Class.cs:69]: This won't show the name of the unknown class anymore (it will say that "name" is an unknown class).

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Comment by spouliot** [comment]
@rolfbjarne anything else needed for approval / merging ?

**Comment by rolfbjarne** [comment]
@spouliot actually yes, there's a monotouch-test failure (on macOS and Mac Catalyst):

```
MonoTouchFixtures.Foundation.NSObjectTest
    [FAIL] SuperClass :   SuperClass
    Expected: 0
    But was:  0x0
    		   at MonoTouchFixtures.Foundation.NSObjectTest.SuperClass() in /Users/builder/azdo/_work/1/s/xamarin-macios/tests/monotouch-test/Foundation/ObjectTest.cs:line 79
```

[execute-Mac-20220217_031707.txt](https://github.com/xamarin/xamarin-macios/files/8117971/execute-Mac-20220217_031707.txt)

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**